Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyComboBox.cxx

```diff
@@ -745,3 +748,15 @@ void qMRMLSubjectHierarchyComboBox::updateComboBoxTitleAndIcon(vtkIdType selecte
     this->setDefaultIcon(QIcon());
     }
 }
+
+//-----------------------------------------------------------------------------
+void qMRMLSubjectHierarchyComboBox::onMRMLSceneCloseEnded(vtkObject* sceneObject)
+{
+  vtkMRMLScene* scene = vtkMRMLScene::SafeDownCast(sceneObject);
+  if (!scene)
+    {
+    return;
+    }
+
+  this->updateComboBoxTitleAndIcon(0);
```

Comment:
Using `vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID` instead of `0` and adding a comment about what it does (e.g., "Update SH combobox title to no selection after scene close") would make things a bit more clear.